pilot/pkg/xds/sds.go

```diff
@@ -299,8 +337,13 @@ func filterAuthorizedResources(resources []SecretResource, proxy *model.Proxy, s
 		if errMessage == nil {
 			errMessage = fmt.Errorf("cross namespace secret reference requires ReferencePolicy")
 		}
-		log.Warnf("proxy %s attempted to access unauthorized certificates %s: %v", proxy.ID, atMostNJoin(deniedResources, 3), errMessage)
+		log.Warn("SDS proxy attempted to access unauthorized certificates",
```

Comment:
I don't think this does what you want. The output will be like `warn    SDS proxy attempted to access unauthorized certificatesproxy1resource2error3proxynamespace4`.

I think you want `log.WithLabels("proxy", proxy.ID, ...).Warn("SDS proxy ....")`

Reply:
Yes, I know. I was looking to fix log.Warn to be closer to slog. After next golang release - I can't see many reasons
to not use the official log API.

Is there any use case for the strange output it generates and not to fix it ?

Reply:
I sent https://github.com/istio/istio/issues/44683 to track slog, and https://github.com/istio/pkg/pull/795 to make it impossible to misuse.

We should not, IMO, **just** make log.Warn behave like  slog. That would mean we have two ways to do the same thing which is confusing. Instead of we should follow https://github.com/istio/istio/issues/44683 and either keep our existing wrapper but use slog under the hood, or move all usage to slog (and drop WithLabels) (preferred)

Reply:
I'm not sure what is wrong with log.Warn behaving like structured logging. There are ~20 uses in the code, most of them would fit very well a more structured style ( or just a fix to add spaces so it's no so strange). 

I'm ok with Warn not generating structured logs - but at least adding space like golang Log seems very simple and
reasonable. I don't know from which NIH project the current WithLabels() style originates ( is it another Istio-only?), but the upstream go and k8s seem to use the simpler style. And if we want to move to golang slog - it would be an easier transition.

Reply:
slog actually does have WithLabels https://pkg.go.dev/golang.org/x/exp/slog#With. Its called With but otherwise identical.

Reply:
Although I guess that means "slog has 2 ways to do the same thing", which was my argument for why we shouldn't do it

Reply:
As a general comment - a lot of our logs are mostly useless for any automated processing and not very tool-friendly. 

An alternative - if you are worried about the few usages of Warn() with no space - is to add a new WarnCtx() ( and similar ) with clear structured semantics, and gradually moving to it. When slog is out we just change the RegisterContext to return a slog, and we can extend a bit the context passed to include common info (proxy, etc).

My longer term goal is to add otel tracing ( ideally a a slog handler to minimize changes) so warnings/errors/etc
can be pushed to an otel sync and better viewed/processed.

Reply:
I don't think there are 2 ways to do the same thing. With() ( and I assume the intent of WithLabels ) would create 
a longer-lived logger with a set of attributes. It has a cost - new object creation, allocating the map, etc - which hopefully can be amortized.  That's what I don't like about WithLabels - even if the log is disabled, it looks like
it still allocates and create the map - just to be Sprinted a couple of times before figuring out if the log was disabled.

log.Info(...args) doesn't require any execution or alloc - and if info is disabled it shouldn't allocate. 

It's not a big deal - I can change the code to use WithLabels and when we adopt slog change it back. 
I'm also not sure if slog is better - or starting to use NewSpan with otel trace library - and having a log
sink for the spans.